manifest/openapi/foundry_v2.go

```diff
@@ -62,26 +64,31 @@ type foapiv2 struct {
 
 // GetTypeByGVK looks up a type by its GVK in the Definitions sections of
 // the OpenAPI spec and returns its (nearest) tftypes.Type equivalent
-func (f *foapiv2) GetTypeByGVK(gvk schema.GroupVersionKind) (tftypes.Type, error) {
+func (f *foapiv2) GetTypeByGVK(gvk schema.GroupVersionKind) (tftypes.Type, map[string]string, error) {
 	f.gate.Lock()
 	defer f.gate.Unlock()
 
+	var hints map[string]string = make(map[string]string)
```

Comment:
That's how I actually started off, with a list. But then I started thinking I'm not 100% sure we won't need to do this for any other type, so I switched to map. 
Opting for one vs the other doesn't reduce the complexity of the whole solution, since most of it lies in just the tedious carrying over of the collection from one end to the other. The map would however reduce the effort next time around, especially for a contributor that's not familiar with the history of this change.
The other important argument (IMO) for the map is that it's much more efficient to search in, given that we have to check each attribute based on their paths. If using a list, we'd have to traverse it for each attribute in turn, comparing it to every entry in the list. That's quite inefficient.